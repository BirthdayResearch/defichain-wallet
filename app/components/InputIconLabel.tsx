import React from 'react'
import { Text } from './Text'
import { tailwind } from '../tailwind'

export enum IconLabelScreenType {
  Balance,
  DEX
}
interface InputIconLabelProps {
  testID?: string
  label: string
  screenType: IconLabelScreenType
}

export function InputIconLabel (props: InputIconLabelProps): JSX.Element {
  return (
    <Text testID={props.testID} style={tailwind('ml-2 text-gray-500 text-right')}>{getSymbolLabel(props.label, props.screenType)}</Text>
  )
}

function getSymbolLabel (symbol: string, screenType: IconLabelScreenType): string {
  let symbolLabel: string

  switch (symbol) {
    case 'DFI':
      symbolLabel = screenType === IconLabelScreenType.Balance ? 'DFI (UTXO)' : 'DFI (Token)'
      break

    default:
      symbolLabel = symbol
  }

  return symbolLabel
}
