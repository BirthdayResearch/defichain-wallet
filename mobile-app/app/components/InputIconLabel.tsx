import { tailwind } from '@tailwind'
import { ThemedText } from './themed'

export enum IconLabelScreenType {
  Balance,
  DEX
}

interface InputIconLabelProps {
  testID?: string
  label: string
  screenType: IconLabelScreenType
}

export function InputIconLabel (props: InputIconLabelProps): JSX.Element {
  return (
    <ThemedText
      dark={tailwind('text-gray-400')}
      light={tailwind('text-gray-500')}
      style={tailwind('ml-2 text-right')}
      testID={props.testID}
    >
      {getSymbolLabel(props.label, props.screenType)}
    </ThemedText>
  )
}

function getSymbolLabel (symbol: string, screenType: IconLabelScreenType): string {
  let symbolLabel: string

  switch (symbol) {
    case 'DFI':
      symbolLabel = screenType === IconLabelScreenType.Balance ? 'DFI (UTXO)' : 'DFI (Token)'
      break

    default:
      symbolLabel = symbol
  }

  return symbolLabel
}
