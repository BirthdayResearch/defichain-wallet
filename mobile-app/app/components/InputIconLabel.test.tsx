import { render } from '@testing-library/react-native'
import { IconLabelScreenType, InputIconLabel } from './InputIconLabel'

jest.mock('@shared-contexts/ThemeProvider')
const screenType = [IconLabelScreenType.Balance, IconLabelScreenType.DEX]

describe('input icon label', () => {
  screenType.forEach(type => {
    it(`should match the output in ${type} screen`, () => {
      const tokenSymbol = 'DFI'
      const component = (
        <InputIconLabel
          label={tokenSymbol}
          screenType={type}
        />
      )
      const rendered = render(component)
      expect(rendered.toJSON()).toMatchSnapshot()
    })
  })
})
