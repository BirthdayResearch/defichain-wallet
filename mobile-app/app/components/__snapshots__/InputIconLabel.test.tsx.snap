// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input icon label should match the output in 0 screen 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      Array [
        Object {
          "marginLeft": 8,
          "textAlign": "right",
        },
        Object {
          "color": "rgba(115, 115, 115, 1)",
        },
      ],
    ]
  }
>
  DFI (UTXO)
</Text>
`;

exports[`input icon label should match the output in 1 screen 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "RegularFont",
        "fontSize": 16,
        "fontWeight": "400",
        "lineHeight": 24,
      },
      Array [
        Object {
          "marginLeft": 8,
          "textAlign": "right",
        },
        Object {
          "color": "rgba(115, 115, 115, 1)",
        },
      ],
    ]
  }
>
  DFI (Token)
</Text>
`;
